### PR TITLE
setup docs

### PR DESCRIPTION
**Pull request by dpo**

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# [Krylov.jl documentation](@id Home)
+
+This package implements iterative methods for the solution of linear systems of equations
+<p align="center">
+  <b><i>Ax = b</i></b>,
+</p>
+and linear least-squares problems
```
Comment (abelsiqueira) [inline docs/src/index.md]: You can use LaTeX ([NLPModels.jl](https://raw.githubusercontent.com/JuliaSmoothOptimizers/NLPModels.jl/master/docs/src/index.md))

File: .travis.yml [hunk 2]
```diff
@@ -18,6 +18,16 @@ branches:
   only:
     - master
```
Comment (abelsiqueira) [inline .travis.yml]: You need this to build on tags (and therefore have docs on tags):
```
- /^v\d+\.\d+(\.\d+)?(-\S*)?$/ # tags
```

**Comment by dpo** [comment]
Any idea why the doc build fails? The error message is absolutely useless.

File: docs/make.jl [hunk 3]
```diff
@@ -0,0 +1,21 @@
+using Documenter, Krylov
+
+makedocs(
+  modules = [Krylov],
+  doctest = true,
+  linkcheck = true,
+  strict = true,
```
Comment (abelsiqueira) [inline docs/make.jl]: This turns the warnings in the build into errors, and that's why it fails.
If you're not gonna document everything right away, better to comment out this line.

Comment (dpo) [reply docs/make.jl]: Thanks. I just found that out. It's ridiculous that it emits warnings and not errors.

**Pull request merged**